packages/hoppscotch-app/newstore/MQTTSession.ts

```diff
@@ -0,0 +1,190 @@
+import { pluck, distinctUntilChanged } from "rxjs/operators"
+import { Client as MQTTClient } from "paho-mqtt"
+import DispatchingStore, { defineDispatchers } from "./DispatchingStore"
+import {
+  HoppRealtimeLog,
+  HoppRealtimeLogLine,
+} from "~/helpers/types/HoppRealtimeLog"
+
+type HoppMQTTRequest = {
+  endpoint: string
+}
+
+type HoppMQTTSession = {
+  request: HoppMQTTRequest
+  connectingState: boolean
+  connectionState: boolean
+  subscriptionState: boolean
+  log: HoppRealtimeLog
+  socket?: MQTTClient
```

Comment:
Do not store the socket because we those are runtime values, do not persist them on disk.

Please refer to `GQLSession.ts` for example.